### ISSUE TITLE
Hide sub-dataverses and files (dataverses that are children of children)

### ISSUE THREAD

**Issue by RightInTwo** [IQSS/dataverse#6549]
As a user, I would like to browse a dataverse installation hierarchically, so that only direct children (dataverses and datasets that belong to THIS dataverse, not to children of THIS dataverse) are displayed.

Motivation: This would help new users understand and find their way around the hierarchy of dataverses.

(Sorry if this has been discussed already, it's kind of hard to find something about it because of the ambiguous search terms....)

**Comment by pdurbin** [comment]
@RightInTwo I don't think this has been discussed but I'm reminded of screenshots in #182 (and somewhat conversations like #5131).

**Comment by RightInTwo** [comment]
I'm a bit surprised that this never came up. Would it be hard to implement a switch like "search within children"?

**Comment by mheppler** [comment]
@RightInTwo it was definitely a design decision to display children and grand/sub children together, in an attempt to make it easier to determine "what is in this". That said, I would be curious to know more about the use case you present, and if anyone else in the community has similar needs and expectations.

My take on your proposed "Sub-Dataverses" facet filter would be to suggest what @pdurbin already hinted at when referencing #5131, that a "Hierarchical" option under the Sort btn could also be a potential solution. Listing direct child first, and so on down the tree.

For the record, hierarchical views of datasets was added in 4.13 (#5572), and I just wanted to reference that work here, as there are a lot of similarities in these two features.

**Comment by RightInTwo** [comment]
> @RightInTwo it was definitely a design decision to display children and grand/sub children together, in an attempt to make it easier to determine "what is in this". That said, I would be curious to know more about the use case you present, and if anyone else in the community has similar needs and expectations.

Well, it confused multiple colleagues of mine. When they add a dataverse to another, they do that to get a better overview, which they don't achieve by nesting dataverses, as everything is still displayed on the top layer.

> My take on your proposed "Sub-Dataverses" facet filter would be to suggest what @pdurbin already hinted at when referencing #5131, that a "Hierarchical" option under the Sort btn could also be a potential solution. Listing direct child first, and so on down the tree.

This would not fulfill my use case, where I want to make the structure of our dataverses easy to understand for newbies. I'd much prefer filtering over sorting.

**Comment by pdurbin** [comment]
> As a user, I would like to browse a dataverse installation hierarchically, so that only direct children (dataverses and datasets that belong to THIS dataverse, not to children of THIS dataverse) are displayed.

It's funny, that's how the "contents" endpoint of the native API works. It only shows direct children. And it show both collections and datasets.

> Motivation: This would help new users to understand and find their way around the hierarchy of dataverses.

Yes, we heard this often. #2155 starts with "Reconsider Featured Dataverses so they could also be used to show the hierarchy of the dataverse." (We'd called these "collections" these days).

@RightInTwo are you still interested in this?

**Comment by RightInTwo** [comment]
Since I am not working with Research Data currently, I'm not pursuing this. Feel free to reopen this issue if need arises.

**Comment by pdurbin** [comment]
Thanks, @RightInTwo . Hopefully we'll create another dataset together in the IQSS kitchen someday!

**Comment by RightInTwo** [comment]
Agreed, @pdurbin! I'll definitely plan a stop if I'm near. That dataset needs some solid follow-up replication anyways! ... and maybe even a larger focus group with an extended universe :D Best wishes to everyone
